tfx/components/transform/executor_test.py

```diff
@@ -58,9 +60,29 @@ class _TempPath(types.Artifact):
 # TODO(b/122478841): Add more detailed tests.
 class ExecutorTest(tft_unit.TransformTestCase):
 
-  def _get_source_data_dir(self):
-    return os.path.join(
+  @classmethod
+  def setUpClass(cls):
+    cls._temp_example_dir = tempfile.mkdtemp()
+    cls._source_data_dir = os.path.join(
         os.path.dirname(os.path.dirname(__file__)), 'testdata')
+    source_example_dir = os.path.join(cls._source_data_dir, 'csv_example_gen')
+
+    cls._artifact1_uri = os.path.join(cls._temp_example_dir, 'csv_example_gen1')
+    io_utils.copy_dir(source_example_dir, cls._artifact1_uri)
+    cls._artifact2_uri = os.path.join(cls._temp_example_dir, 'csv_example_gen2')
+    io_utils.copy_dir(source_example_dir, cls._artifact2_uri)
+
+    # Duplicate the number of train and eval records such that
+    # second artifact has twice as many as first.
+    artifact2_train_dir = os.path.join(cls._artifact2_uri, 'train')
+    for filename in tf.io.gfile.listdir(artifact2_train_dir):
+      io_utils.copy_file(os.path.join(artifact2_train_dir, filename),
+                         os.path.join(artifact2_train_dir, 'dup_' + filename))
+
+    artifact2_eval_dir = os.path.join(cls._artifact2_uri, 'eval')
+    for filename in tf.io.gfile.listdir(artifact2_eval_dir):
+      io_utils.copy_file(os.path.join(artifact2_eval_dir, filename),
+                         os.path.join(artifact2_eval_dir, 'dup_' + filename))
```

Comment:
you can just glob all files under cls._artifact2_uri and use one for loop to do duplication I think

and since train:eval = 2:1, the artifact1:artifact2= 2:1, will artifact1train = artifact2eval, or it's slightly different

Reply:
i think its slightly different, the numbers are below

artifact1 train 9909
artifact1 eval 5091
artifact2 train 19818
artifact2 eval 10182

Reply:
updated the behavior to use a single glob